src/cmd/tag.go

```diff
@@ -0,0 +1,139 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/opslevel/cli/common"
+	"github.com/opslevel/opslevel-go/v2023"
+	"golang.org/x/exp/slices"
+
+	"github.com/spf13/cobra"
+)
+
+var resourceType string
+
+var createTagCmd = &cobra.Command{
+	Use:   "tag --type=RESOURCE_TYPE [--assign] RESOURCE_ID KEY VALUE",
+	Short: "Create/assign a tag",
+	Long:  "Create/assign a tag",
+	Example: `
+opslevel create tag --type=Team ID|ALIAS KEY VALUE
+`,
+	Args:       cobra.ExactArgs(3),
+	ArgAliases: []string{"RESOURCE_ID", "KEY", "VALUE"},
+	Run: func(cmd *cobra.Command, args []string) {
+		if resourceType == "" {
+			err := errors.New("must specify a taggable resource type using --type=RESOURCE_TYPE")
+			cobra.CheckErr(err)
+		}
+		if !slices.Contains(opslevel.AllTaggableResource, resourceType) {
+			err := errors.New("not a taggable resource type: " + resourceType)
+			cobra.CheckErr(err)
+		}
+
+		resource := args[0]
+		key := args[1]
+		value := args[2]
+
+		if cmd.Flag("assign").Changed {
+			var input opslevel.TagAssignInput
+			tagInput := opslevel.TagInput{
+				Key:   key,
+				Value: value,
+			}
+
+			if opslevel.IsID(resource) {
+				input = opslevel.TagAssignInput{
+					Id:   opslevel.ID(resource),
+					Tags: []opslevel.TagInput{tagInput},
+				}
+			} else {
+				input = opslevel.TagAssignInput{
+					Alias: resource,
+					Type:  opslevel.TaggableResource(resourceType),
+					Tags:  []opslevel.TagInput{tagInput},
+				}
+			}
```

Comment:
What do you think about this approach?
```suggestion
			tagInput := opslevel.TagInput{
				Key:   key,
				Value: value,
			}
      input := opslevel.TagAssignInput{Tags: []opslevel.TagInput{tagInput}}

			if opslevel.IsID(resource) {
				input.Id = opslevel.ID(resource)
			} else {
				input.Alias = resource
				input.Type = opslevel.TaggableResource(resourceType)
			}

```